mycpp/control_flow_pass.py

```diff
@@ -0,0 +1,418 @@
+        pass
+
+    def visit_await_expr(self, o: 'mypy.nodes.AwaitExpr') -> T:
+        pass
+
+    def visit_temp_node(self, o: 'mypy.nodes.TempNode') -> T:
+        pass
+
+    def visit_assignment_stmt(self, o: 'mypy.nodes.AssignmentStmt') -> T:
+        pass
+
+    def visit_for_stmt(self, o: 'mypy.nodes.ForStmt') -> T:
+        cfg = self.current_cfg()
+        if not cfg:
+            return
+
+        with pass_state.CfgLoopContext(cfg) as loop:
+            self.loop_stack.append(loop)
+            self.accept(o.body)
+            self.loop_stack.pop()
+
+    def _handle_switch(self, expr, o, cfg):
+        assert len(o.body.body) == 1, o.body.body
+        if_node = o.body.body[0]
+        assert isinstance(if_node, IfStmt), if_node
+        cases = []
+        default_block = util._collect_cases(self.module_path, if_node, cases)
+        with pass_state.CfgBranchContext(cfg, self.current_statement_id) as branch_ctx:
+            for expr, body in cases:
+                assert expr is not None, expr
+                with branch_ctx.AddBranch():
+                    self.accept(body)
+
+            if default_block:
+                with branch_ctx.AddBranch():
+                    self.accept(default_block)
+
+    def visit_with_stmt(self, o: 'mypy.nodes.WithStmt') -> T:
+        cfg = self.current_cfg()
+        if not cfg:
+            return
+
+        assert len(o.expr) == 1, o.expr
+        expr = o.expr[0]
+        assert isinstance(expr, CallExpr), expr
+
+        callee_name = expr.callee.name
+        if callee_name == 'switch':
+            self._handle_switch(expr, o, cfg)
+        elif callee_name == 'str_switch':
+            self._handle_switch(expr, o, cfg)
+        elif callee_name == 'tagswitch':
+            self._handle_switch(expr, o, cfg)
+        else:
+            with pass_state.CfgBlockContext(cfg, self.current_statement_id):
+                for stmt in o.body.body:
+                    self.accept(stmt)
+
+    def visit_del_stmt(self, o: 'mypy.nodes.DelStmt') -> T:
+        pass
+
+    def visit_func_def(self, o: 'mypy.nodes.FuncDef') -> T:
+        if o.name == '__repr__':  # Don't translate
+            return
+
+        self.cfgs[split_py_name(o.fullname)] = pass_state.ControlFlowGraph()
+        self.current_func_node = o
+        self.accept(o.body)
+        self.current_func_node = None
+
+    def visit_overloaded_func_def(self,
+                                  o: 'mypy.nodes.OverloadedFuncDef') -> T:
+        pass
+
+    def visit_class_def(self, o: 'mypy.nodes.ClassDef') -> T:
+        for stmt in o.defs.body:
+            # Ignore things that look like docstrings
+            if (isinstance(stmt, ExpressionStmt) and
+                    isinstance(stmt.expr, StrExpr)):
+                continue
+
+            if isinstance(stmt, FuncDef) and stmt.name == '__repr__':
+                continue
+
+            self.accept(stmt)
+
+    def visit_global_decl(self, o: 'mypy.nodes.GlobalDecl') -> T:
```

Comment:
I think we can delete all this stuff, i.e. all the empty methods

But yeah the other passes are a mess too, so we can do that all at once

I think I wasn't sure what "dialect" of Python we were using until the translation was done, so I kept them all "just in case"

But now it's done so we can go back and gradually clean things up

Reply:
oh nice! i wasn't sure what the defaults for `ExpressionVisitor` and `StatementVisitor` were. let's drop these for now